service/history/workflow/mutable_state.go

```diff
@@ -292,7 +292,7 @@ type (
 		ApplyWorkflowExecutionUpdateCompletedEvent(event *historypb.HistoryEvent, batchID int64) error
 		SetCurrentBranchToken(branchToken []byte) error
 		SetHistoryBuilder(hBuilder *historybuilder.HistoryBuilder)
-		SetHistoryTree(executionTimeout *durationpb.Duration, runTimeout *durationpb.Duration, treeID string) error
+		SetHistoryTree(shardID int32, executionTimeout *durationpb.Duration, runTimeout *durationpb.Duration, treeID string) error
```

Comment:
this is the MutableState interface ... shouldn't we expect the mutable state to always (already) know what shard it lives in?